README.md

```diff
@@ -0,0 +1,198 @@
+# Test initialization.
+@setUpServer()
+@setUp()
+@setUpClient()
+# Test body.
+@testFoo()
+# Test cleanup.
+@tearDownClient()
+@tearDownServer()
+@tearDown()
+```
+
+Server-side and client-side tests
+---------------------------------
+
+By default all tests run both on client and server. It is possible to specify that some should only be executed on either the server-side or the client-side. This is done through a method naming convention which is as follows:
+
+ * If a method name begins with `testServer` then the test will only be executed on the server side.
+ * If a method name begins with `testClient` then the test will only be executed on the client side.
+
+Asynchronous tests
+------------------
+
+Tests can be specified in two ways:
+
+ * A single test method as in the above examples. When such a test method finishes, the test is deemed complete and the respective tear down methods will run.
+ * Test containing multiple steps where each step is only deemed complete after certain callbacks get called. This is similar to `testAsyncMulti` from `test-helpers` (actually `testAsyncMulti` is used in the background to make this work).
+
+In the second case, the test should not be defined as a method, but rather as an array of functions like in the following example:
+
+```coffeescript
+  testClientFoo:
+    [
+      ->
+        # Call the first method.
+        Meteor.call 'first', 'argument', @expect (error, result) =>
+          @assertIsUndefined error, "Error while calling first: #{ error }"
+    ,
+      ->
+        # Call the second method.
+        Meteor.call 'second', 'argument', @expect (error, result) =>
+          @assertIsUndefined error, "Error while calling second: #{ error }"
+    ]
+```
+
+This defines a chain of sub-tests where the next case will only get executed once all the *expected* callbacks are run. In order to define which callbacks are expected one should use the `@expect(fun)` method which takes a function argument and returns a wrapper function that will mark the callback as called. When all expected callbacks are called, the execution will proceed to the next sub-test in the chain.
+
+Note that in this case set up and tear down methods are only called once for the whole test and not in-between sub-tests.
+
+Interleaving client-side and server-side assertions
+--------------------------------------------------
+
+Sometimes it can be useful to first run some tests on the client, then after those are done, run some tests on the server to check whether the client calls correctly affected the backend storage. This can be done by interleaving client-side sub-tests with server-side sub-tests. We take the previous async test example and add a server-side sub-test between the existing two:
+
+```coffeescript
+  testClientFoo:
+    [
+      ->
+        # Call the first method.
+        Meteor.call 'first', 'argument', @expect (error, result) =>
+          @assertIsUndefined error, "Error while calling first: #{ error }"
```

Comment:
BTW, I am not sure if it is the best thing to test for undefined here. At least I am passing null as `error`, when there is no error. (Because I do not want to use `undefined` in my own code.)

In my tests I was simply testing if `error` is false. I think that is a node.js callback style, no? Or should it be `undefined`? Can you check?